edk2toolext/edk2_invocable.py

```diff
@@ -104,18 +106,42 @@ def ParseCommandLineOptions(self):
             self.PlatformModule = import_module_by_file_name(os.path.abspath(settingsArg.platform_module))
             self.PlatformSettings = locate_class_in_module(
                 self.PlatformModule, self.GetSettingsClass())()
-        except (TypeError, FileNotFoundError) as e:
-            # Gracefully exit if setup doesn't go well.
+        except (TypeError):
+            # Gracefully exit if the file we loaded isn't the right type
+            class_name = self.GetSettingsClass().__name__
+            print(f"Unable to use {settingsArg.platform_module} as a {class_name}")
+            print("Did you mean to use a different kind of invokable?")
             try:
                 # If this works, we can provide help for whatever special functions
                 # the subclass is offering.
                 self.AddCommandLineOptions(settingsParserObj)
+                Module = self.PlatformModule
+                module_contents = dir(Module)
+                # Filter through the Module, we're only looking for classes.
+                classList = [getattr(Module, obj) for obj in module_contents if inspect.isclass(getattr(Module, obj))]
+                # Get the name of all the classes
+                classNameList = [obj.__name__ for obj in classList if is_instance(obj)]
+                # TODO improve is_instance as it catches imports as well as declared classes
+                imported_classes = ", ".join(classNameList)  # Join the classes together
+                print(f"The module you imported contains {imported_classes}")
```

Comment:
can these print statements not be done using the logging module?  logging.critical seems like it would be better here unless this happens before logging is up?

Reply:
We don't have logging setup up yet, so we can't do logging.

Reply:
We need the platform settings to get the loglevel and settings